modules/pam_mkhomedir/mkhomedir_helper.c

```diff
@@ -153,15 +155,17 @@ create_homedir(const struct passwd *pwd, mode_t dir_mode,
 	 if (pointedlen >= 0) {
             if(symlink(pointed, newdest) != 0)
             {
-               pam_syslog(NULL, LOG_DEBUG,
-			  "unable to create link %s: %m", newdest);
+               retval = errno = EEXIST ? PAM_SUCCESS : PAM_PERM_DENIED;
```

Comment:
`errno = EEXIST` is an assignment, not a comparison.

Reply:
Fixed as well